KerbalStuff/app.py

```diff
@@ -70,49 +69,77 @@ def load_user(username: str) -> User:
-            return handle_api_exception(e)
-        return render_template("internal_error.html"), 500
-
-    # Error handler
-    error_to = _cfg("error-to")
-    if error_to:
-        import logging
-        from logging.handlers import SMTPHandler
-
-        smtp_host = _cfg('smtp-host')
-        smtp_port = _cfgi('smtp-port')
-        error_from = _cfg('error-from')
-        site_name = _cfg('site-name')
-        if smtp_host and smtp_port and error_from and site_name:
-            mail_handler = SMTPHandler((smtp_host, smtp_port),
-                                       error_from, [error_to],
-                                       site_name + ' Application Exception')
-            mail_handler.setLevel(logging.ERROR)
-            app.logger.addHandler(mail_handler)
+# Send ERROR and EXCEPTION level log entries to syslog and per email.
+error_to = _cfg("error-to")
+if error_to:
+    import logging
+    from logging.handlers import SMTPHandler
+
+    smtp_host = _cfg('smtp-host')
+    smtp_port = _cfgi('smtp-port')
+    error_from = _cfg('error-from')
+    site_name = _cfg('site-name')
+    if smtp_host and smtp_port and error_from and site_name:
+        mail_handler = SMTPHandler((smtp_host, smtp_port),
+                                   error_from, [error_to],
+                                   site_name + ' Application Exception')
+        mail_handler.setLevel(logging.ERROR)
+        app.logger.addHandler(mail_handler)
+
+# Error handlers. We want to handle errors differently based on their nature, whether it's done via XHR,
+# and whether we're in a debug/development environment.
+
+# app.debug | XHR | HTTPException | What to do?
+# ---------------------------------------------
+# 0         | 0   | 0             | Real 500 -> log, db rollback, error_500.html
+# 0         | 0   | 1             | 4XX -> error_404.html or error_4XX.html
+# 0         | 1   | 0             | Real 500 in API -> log, db rollback, jsonified_exception(e)
+# 0         | 1   | 1             | 4XX in API -> jsonified_exception(e)
+# 1         | 0   | 0             | Real 500 -> log, db rollback, debugger
+# 1         | 0   | 1             | 4XX -> error_404.html or error_4XX.html
+# 1         | 1   | 0             | Real 500 in API -> log, db rollback, jsonified_exception(e)
+# 1         | 1   | 1             | 4XX in API -> jsonified_exception(e)
 
 
 @app.errorhandler(404)
-def handle_404(e: Exception) -> Union[Tuple[str, int], werkzeug.wrappers.Response]:
-    path = request.path
-    if path.startswith('/api/'):
-        return handle_api_exception(e)
-    return render_template("not_found.html"), 404
+def handle_404(e: HTTPException) -> Union[Tuple[str, int], werkzeug.wrappers.Response]:
+    # Switch out the default message
+    if e.description == werkzeug.exceptions.NotFound.description:
+        e.description = "Requested page not found. Looks like this was deleted, or maybe was never here."
+
+    if request.path.startswith("/api") \
```

Comment:
```suggestion
    if request.path.startswith("/api/") \
```

Let's guard against this being used for any future `/apiary` or `/apices` routes. :wink: